### ISSUE TITLE
Bad handling of internal long doc comments

### ISSUE THREAD

**Issue by Ms2ger** [rust-lang-nursery/rustfmt#553]
```diff
 fn foo() {
-    /*!
-     * This does foo stuff.
-     */
+    // !
+    // This does foo stuff.
+    //
 }
```

**Comment by crumblingstatue** [comment]
I'm interested in eventually running rustfmt on [rust-sfml](https://github.com/jeremyletang/rust-sfml), but this bug causes missing doc comments for various items.

```Diff
-/*!
- * Base module of SFML, defining various utilities.
- *
- * It provides vector classes, unicode strings and conversion functions, threads and mutexes, timing classes.
- */
+// !
+// Base module of SFML, defining various utilities.
+//
+// It provides vector classes, unicode strings and conversion functions, threads and mutexes, timing classes.
+//

```

**Comment by nrc** [comment]
Also happens with `/**` doc comments (why do we have so many kinds of doc comments?!):

```
-    /**
-    Is the package directory in the cache?
+// *
+// Is the package directory in the cache?
+//
+// Currently, this can be inferred from `emit_metadata`, but there's no *intrinsic* reason they should be tied together.
+//
```

**Comment by nrc** [comment]
I believe this is mostly fixed by #681, and that the remaining issues are covered by other issues.
